odoo/orm/fields.py

```diff
@@ -1107,6 +1121,197 @@ def update_db_related(self, model):
             join_field=SQL.identifier(join_field),
         ))
 
+    ############################################################################
+    #
+    # SQL generation methods
+    #
+
+    def field_to_sql(self, model: BaseModel, alias: str, flush: bool = True) -> SQL:
```

Comment:
The method name is awkward: `field.field_to_sql()`.  Was it necessary to put that in the same PR?  This PR covers more stuff than necessary IMHO.